salt/states/file.py

```diff
@@ -747,18 +748,18 @@ def _check_directory(name,
                         fchange['user'] = user
                     if group is not None and group != stats.get('group'):
                         fchange['group'] = group
-                    if mode is not None and mode != stats.get('mode'):
-                        fchange['mode'] = mode
+                    if file_mode is not None and file_mode != stats.get('mode'):
```

Comment:
the "mode" kwarg got changed to "dir_mode", so shouldn't this be "dir_mode" instead of "file_mode"?